pkglistgen/tool.py

```diff
@@ -623,70 +685,126 @@ def update_and_solve_target(
+                os.makedirs(cache_dir)
 
         group_dir = os.path.join(cache_dir, group)
         product_dir = os.path.join(cache_dir, product)
+        self.productcompose_dir = os.path.join(cache_dir, productcompose)
         release_dir = os.path.join(cache_dir, release)
         oldrepos_dir = os.path.join(cache_dir, oldrepos)
 
         self.input_dir = group_dir
         self.output_dir = product_dir
 
-        for package in checkout_list:
-            if no_checkout:
-                logging.debug('Skipping checkout of {}/{}'.format(project, package))
-                continue
-            checkout_package(api.apiurl, project, package, expand_link=True,
-                             prj_dir=cache_dir, outdir=os.path.join(cache_dir, package))
+        # validating that an old product-builder product is used at all
+        if os.path.isdir(product_dir):
+            file_utils.unlink_all_except(release_dir, ['weakremovers.inc'])
+        else:
+            # otherwise just unset product_dir to skip all product-builder actions
+            product_dir = None
+            self.output_dir = self.productcompose_dir
+
+        if not no_checkout and not git_url:
+            logging.debug('Skipping checkout of {}'.format(project))
+            for package in checkout_list:
+                checkout_package(api.apiurl, project, package, expand_link=True,
+                                 prj_dir=cache_dir, outdir=os.path.join(cache_dir, package))
+
 
-        file_utils.unlink_all_except(release_dir, ['weakremovers.inc', '*.changes'])
-        if not only_release_packages:
-            file_utils.unlink_all_except(product_dir)
-        ignore_list = ['supportstatus.txt', 'summary-staging.txt', 'package-groups.changes']
+        ignore_list = ['supportstatus.txt', 'summary-staging.txt', 'package-groups.changes',
+                       'default.productcompose.in']
         ignore_list += self.group_input_files()
-        file_utils.copy_directory_contents(group_dir, product_dir, ignore_list)
-        file_utils.change_extension(product_dir, '.spec.in', '.spec')
-        file_utils.change_extension(product_dir, '.product.in', '.product')
+
+        # old product-builder
+        if product_dir:
+           if not only_release_packages:
+               file_utils.unlink_all_except(product_dir)
+           file_utils.copy_directory_contents(group_dir, product_dir, ignore_list)
+           file_utils.change_extension(product_dir, '.spec.in', '.spec')
+           file_utils.change_extension(product_dir, '.product.in', '.product')
+           file_utils.unlink_all_except(release_dir, ['weakremovers.inc', '*.changes'])
+           if not only_release_packages:
+               file_utils.unlink_all_except(product_dir)
+
+        # new product-composer
+        fn = os.path.join(group_dir, 'default.productcompose.in')
+        if os.path.isfile(fn):
+            if not os.path.isdir(self.productcompose_dir):
```

Comment:
IMO package creation should be up to the maintainer as a one-time action

Reply:
it is an exception now.